### PR TITLE
Bug/464 unique torch tensor

### PR DESCRIPTION
**Pull request by ClaudiaComito**
## Description

Output of ht.unique(x) used to be a torch tensor or a tuple of torch tensors if `x.split is None`, it is now either a heat tensor, or a tuple of heat tensors in all cases.

Issue/s resolved: #464 

## Changes proposed:
- see above
- tests updated

## Type of change

Remove irrelevant options:
- Bug fix (non-breaking change which fixes an issue)

## Due Diligence

- [x] All split configurations tested
- [x] Multiple dtypes tested in relavent functions
- [x] Documentation updated (if needed)
- [x] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
 no

### CODE REVIEW

File: heat/core/manipulations.py [hunk 1]
```diff
@@ -1036,10 +1036,14 @@ def unique(a, sorted=False, return_inverse=False, axis=None):
            [3, 1]])
     """
     if a.split is None:
-        # Trivial case, result can just be forwarded
-        return torch.unique(
+        torch_output = torch.unique(
             a._DNDarray__array, sorted=sorted, return_inverse=return_inverse, dim=axis
         )
+        if isinstance(torch_output, tuple):
+            heat_output = tuple(factories.array(i, dtype=a.dtype) for i in torch_output)
```
Comment (mtar) [inline heat/core/manipulations.py]: different number of arguments?

File: heat/core/manipulations.py [hunk 2]
```diff
@@ -1036,10 +1036,14 @@ def unique(a, sorted=False, return_inverse=False, axis=None):
...
+        else:
+            heat_output = factories.array(torch_output, dtype=a.dtype, split=None, device=a.device)
```
Comment (mtar) [inline heat/core/manipulations.py]: different number of arguments?

File: heat/core/manipulations.py (hunk 1 above)
Comment (coquelin77) [reply heat/core/manipulations.py]: also the array calls should have a device parameter

Comment (coquelin77) [reply heat/core/manipulations.py]: and split

**Comment by ClaudiaComito** [comment]
Thanks a lot @mtar and @coquelin77 , you're correct, I made the changes you requested, merging and closing now.

**Pull request merged**